sdcm/nemesis.py

```diff
@@ -292,6 +294,23 @@ def wrapper(self, *args, **kwargs):
         setattr(cls, func.__name__, wrapper)  # bind it to Nemesis class
         return func  # returning func means func can still be used normally
 
+    @staticmethod
+    @contextmanager
+    def run_nemesis(node_list: list['BaseNode'], nemesis_label: str):
+        """
+        pick a node out of a `node_list`, and mark is as running_nemesis
+        for the duration of this context
+        """
+        with NEMESIS_TARGET_SELECTION_LOCK:
+            free_nodes = [node for node in node_list if not node.running_nemesis]
+            node = random.choice(free_nodes)
```

Comment:
Shouldn't it have check for emptiness?

Reply:
we can, check and raise error, anyhow error would be raise if list would be empty :)

Reply:
If we can raise expected error then why not?

Reply:
fixed it, and now we do